src/meter/Meter.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import FlexView from '../flex/FlexView';
+import find from 'lodash/collection/find';
+import { warn } from '../utils/log';
+
+const Range = t.refinement(t.struct({
+  startValue: t.maybe(t.Number),
+  endValue: t.Number,
+  fillingColor: t.String,
+  labelColor: t.String
+}), r => r.startValue < r.endValue);
+
+const labelFormatter = (currentValue, startValue, endValue) =>(
+  Math.abs((currentValue - startValue) * 100 / (endValue - startValue))+'%'
+);
+/**
+ * ### Renders a Progress Bar
+ */
+@skinnable()
+@props({
+  /**
+   * This is the value provided as input.
+   */
+  currentValue: t.Number,
+  /**
+   * Minimum value.
+   */
+  minValue: t.maybe(t.Number),
+  /**
+   * Maximum value.
+   */
+  maxValue: t.maybe(t.Number),
+  /**
+   * Function as input in which you can define a custom label format.
+   */
+  labelFormatter: t.maybe(t.Function),
+  /**
+   * Array of Object in which you can define startValue, endValue, labelColor, fillingColor.
+   */
+  steps: t.maybe(t.list(Range)),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+})
+export default class Meter extends React.Component{
+
+  static defaultProps = {
+    minValue: 0,
+    maxValue: 100,
+    labelFormatter
+  };
+
+  validateSteps = (steps) => {
+    if(!steps){
+      return false;
+    } else {
+      return steps.reduce(( memo, curr, index, arr ) => {
+        if(index === 1){
+          memo.startValue = this.props.minValue;
+        }
+        curr.startValue = arr[index-1].endValue + 1; //scale to be defined
+        return memo;
+      });
+    }
+  }
+
+  getSteps = () => {
+    const {
+      steps,
+      minValue,
+      maxValue
+    } = this.props;
+
+    if(this.validateSteps(steps)){
+      console.log("steps =>", steps);
+      return steps;
+    }
+    return [{
+      startValue: minValue,
+      endValue: maxValue,
+      fillingColor: '#ccc',
+      labelColor: '#000'
+    }];
+  }
+
+  getResultFromCurrentValue = ({ currentValue, minValue, maxValue }) => {
+    return Math.abs((currentValue - minValue) * 100 / (maxValue - minValue));
+  }
+
+  getStyles = () => {
+    const {
+      currentValue,
+      minValue,
+      maxValue
+    } = this.props;
+    const step = find(this.getSteps(), ({ startValue, endValue })  => {
+      return currentValue >= startValue && currentValue <= endValue; //To be checked, allow overlapping atm
+    });
+    const result = this.getResultFromCurrentValue({ currentValue, minValue, maxValue });
+    return {
+      wrapperStyle: {
+        width: result+'%',
+        height: '100%',
+        backgroundColor: step.fillingColor || '#ccc'
+      },
+      labelStyle: {
+        color: step.labelColor || '#000'
+      }
```

Comment:
you should honor the default label color and default filling color props here, which I see are not part of the props api yet..